substrate/frame/assets/src/functions.rs

```diff
@@ -683,6 +717,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				source_died =
 					Some(Self::dead_account(source, details, &source_account.reason, false));
 				if let Some(Remove) = source_died {
+					Self::ensure_account_can_die(id.clone(), source)?;
```

Comment:
This does return an error after the modification of storage `Account` above.

Reply:
Done on [`d6af473`](https://github.com/paritytech/polkadot-sdk/pull/4530/commits/d6af4735770f902f10f92c69fc03945462f4ac2a).

Reply:
You can review again @gui1117